lib/matplotlib/axes/_base.py

```diff
@@ -4383,12 +4385,20 @@ def get_tightbbox(self, renderer, call_axes_locator=True,
             bbox_artists = self.get_default_bbox_extra_artists()
 
         for a in bbox_artists:
-            bbox = a.get_tightbbox(renderer)
-            if (bbox is not None
-                    and 0 < bbox.width < np.inf
-                    and 0 < bbox.height < np.inf):
-                bb.append(bbox)
-
+            # Extra check here to quickly see if clipping is on and
+            # contained in the axes.  If it is, don't get the tightbbox for
+            # this artist because this can be expensive:
+            clip_extent = a._get_clipping_extent_bbox()
+            if clip_extent is not None:
+                clip_extent = mtransforms.Bbox.intersection(clip_extent,
+                    axbbox)
+                inside_axes = np.all(clip_extent.extents == axbbox.extents)
```

Comment:
if not inside_axes: continue

and then below you don't need to re-check for Noneness of clip_extent.

Reply:
Refactored to make this cleaner...

Reply:
... though I have a mild aversion to `continue`.  Somewhere I got accustomed to `continue` being bad style...